form/create_custom_choice_type.rst

```diff
@@ -0,0 +1,273 @@
+:doc:`EntityType </reference/forms/types/entity>` which loads the choices from
+a set of Doctrine entities.
+
+It's also common to want to re-use the same list of choices for different fields.
+Creating a custom "choice" field is a great solution - something like::
+
+    use App\Form\Type\CategoryChoiceType;
+
+    // ... from any type
+    $builder
+        ->add('category', CategoryChoiceType::class, [
+            // ... some inherited or custom options for that type
+        ])
+        // ... add more fields
+    ;
+
+
+Creating a Type With Static Custom Choices
+------------------------------------------
+
+To create a custom choice type when choices are static, you can do the
+following::
+
+    // src/Form/Type/CategoryChoiceType.php
+    namespace App\Form\Type;
+
+    use App\Domain\Model;
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    class CategoryChoiceType extends AbstractType
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function getParent()
+        {
+            // inherits all options, form and view configuration
+            // to create expanded or multiple choice lists
+            return ChoiceType::class;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            $resolver
+                // Using whatever way you want to get the choices - Mode::getCategories() is just an example
+                ->setDefault('choices', Model::getCategories())
+
+                // ... override more choice options or define new ones
+            ;
+        }
+    }
+
+.. caution::
+
+     The ``getParent()`` method is used instead of ``extends``.
+     This allows the type to inherit from both ``FormType`` and ``ChoiceType``.
+
+Loading Lazily Static Custom Choices
+------------------------------------
+
+Sometimes, the callable to define the ``choices`` option can be a heavy process
+that could be prevented when the submitted data is optional and empty.
+Sometimes it can depend on other options.
+
+The solution is to load the choices lazily using the ``choice_loader`` option,
+which accepts a callback::
+
+    use Symfony\Component\Form\ChoiceList\ChoiceList;
+    use Symfony\Component\OptionsResolver\Options;
+
+    // use this option instead of the "choices" option
+    ->setDefault('choice_loader', ChoiceList::lazy($this, static function() {
+        return Model::getCategories();
+    }))
+
+    // or if it depends on other options
+    ->setDefault('some_option', 'some_default')
+    ->setDefault('choice_loader', static function (Options $options) {
+        $someOption = $options['some_option'];
+
+        return ChoiceList::lazy($this, static function() use ($someOption) {
+            return Model::getCategories($someOption);
+        }, $someOption);
+    }))
```

Comment:
I would always make sure that our code examples are syntax correct (because it's often easier to read - you know how to call the method - and the Pygments code highlighter is sometimes very strict in these things - making the example completely read if there is something wrong).

So in this code example,  please add `$resolver` to each call

Reply:
Fixed thanks!